scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1888,3 +1888,338 @@ cdef class NaiveRatioUniforms(Method):
+            unur_par *par
+            unur_gen *rng
+
+        self._messages = MessageStream()
+        _lock.acquire()
+
+        try:
+            unur_set_stream(self._messages.handle)
+
+            self.distr = unur_distr_discr_new()
+
+            if self.distr == NULL:
+                raise UNURANError(self._messages.get())
+
+            n_pv = len(pv_view)
+            self._check_errorcode(unur_distr_discr_set_pv(self.distr, &pv_view[0], n_pv))
+
+            if domain is not None:
+                self._check_errorcode(unur_distr_discr_set_domain(self.distr, domain[0], domain[1]))
+
+            self.par = unur_dgt_new(self.distr)
+            if self.par == NULL:
+                raise UNURANError(self._messages.get())
+
+            self._check_errorcode(unur_dgt_set_guidefactor(self.par, guide_factor))
+            self._set_rng(random_state)
+        finally:
+            _lock.release()
+
+    cdef object _validate_args(self, dist, domain, guide_factor):
+        cdef double[::1] pv_view
+
+        domain = _validate_domain(domain, dist)
+        if domain is not None:
+            if not np.isfinite(domain).all():
+                raise ValueError("`domain` must be finite.")
+        else:
+            if hasattr(dist, 'pmf'):
+                raise ValueError("`domain` must be provided when the "
+                                 "probability vector is not available.")
+
+        if guide_factor > 3:
+            msg = "guide_factor sizes larger than 3 are not recommended."
+            warnings.warn(msg, RuntimeWarning)
+
+        if guide_factor == 0:
+            msg = ("If the relative size (guide_factor) is set to 0, "
+                   "sequential search is used. However, this is not "
+                   "recommended, except in exceptional cases, since the "
+                   "discrete sequential search method has almost no setup and "
+                   "is thus faster.")
+            warnings.warn(msg, RuntimeWarning)
+
+        if hasattr(dist, 'pmf'):
+            # we assume the PMF accepts and return floats. So, we need
+            # to vectorize it to call with an array of points in the domain.
+            pmf = np.vectorize(dist.pmf)
+            k = np.arange(domain[0], domain[1]+1)
+            pv = pmf(k)
+            try:
+                pv_view = _validate_pv(pv)
+            except ValueError as err:
+                msg = "PMF returned invalid values: " + err.args[0]
+                raise ValueError(msg) from None
+        else:
+            pv_view = _validate_pv(dist)
+
+        return pv_view, domain
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef inline void _ppf(self, const double *u, double *out, size_t N):
+        cdef:
+            size_t i
+        for i in range(N):
+            out[i] = unur_dgt_eval_invcdf(self.rng, u[i])
+
+    def ppf(self, u):
+        """
+        ppf(u)
+
+        Approximated PPF of the given distribution.
```

Comment:
remove Approximated since DGT returns the exact value

Reply:
Done.